src/dune_pkg/opam_solver.ml

```diff
@@ -237,6 +237,20 @@ let opam_command_to_string_debug (args, _filter_opt) =
   |> String.concat ~sep:" "
 ;;
 
+let package_variable_of_opam_ident package ident =
+  let err_not_supported var package =
+    User_error.raise
+      [ Pp.textf
+          "Variable %S occuring in opam package %S is not supported."
+          var
+          (OpamPackage.to_string package)
+      ]
+  in
+  match Package_variable.pform_of_opam_ident ident with
```

Comment:
What about just moving the check for `"root"` here. That would allow us not to modify the type.

Reply:
I would have to pass the package name. Would that be acceptable?

Reply:
I think so. My thinking is that this value of `Pform` can't be used correctly under any circumstances. So no need to allow people to build values from it.